exec/vector/src/main/java/org/apache/drill/exec/vector/DateUtilities.java

```diff
@@ -213,11 +216,27 @@ public static LocalDate fromDrillDate(long value) {
     return LocalDate.ofEpochDay(value / daysToStandardMillis);
   }
 
-  public static long toDrillTimestamp(Instant instant) {
-    return instant.toEpochMilli();
+  /**
+   * Convert to seconds since the _local_ Unix epoch, midnight 1970-01-01
+   * in the configured time zone.
+   * @param instant an Instant representing a moment in time.
+   * @param zoneId Drill's configured time zone.
+   * @return the Drill timestamp value corresponding to instant.
+   */
+  public static long toDrillTimestamp(Instant instant, ZoneId zoneId) {
+    LocalDateTime ldt = instant.atZone(zoneId).toLocalDateTime();
+    return ldt.atZone(ZoneId.of("UTC")).toEpochSecond()*1000;
```

Comment:
1) I think we should use the `toInstant().toEpochMilli()`. Current implementation loses information about milliseconds.
2) `ZoneId.of ("UTC")` is a good candidate for creating a constant

Reply:
Good catch, thanks.